src/node_zlib.cc

```diff
@@ -1253,6 +1434,109 @@ CompressionError BrotliDecoderContext::GetErrorInfo() const {
   }
 }
 
+void ZstdContext::Close() {}
+
+void ZstdContext::SetBuffers(const char* in,
+                             uint32_t in_len,
+                             char* out,
+                             uint32_t out_len) {
+  input_.src = reinterpret_cast<const uint8_t*>(in);
+  input_.size = in_len;
+  input_.pos = 0;
+
+  output_.dst = reinterpret_cast<uint8_t*>(out);
+  output_.size = out_len;
+  output_.pos = 0;
+}
+
+void ZstdContext::SetFlush(int flush) {
+  flush_ = static_cast<ZSTD_EndDirective>(flush);
+}
+
+void ZstdContext::GetAfterWriteOffsets(uint32_t* avail_in,
+                                       uint32_t* avail_out) const {
+  *avail_in = input_.size - input_.pos;
+  *avail_out = output_.size - output_.pos;
+}
+
+CompressionError ZstdContext::GetErrorInfo() const {
+  if (error_ != ZSTD_error_no_error) {
+    return CompressionError(error_string_.c_str(),
+                            error_code_string_.c_str(),
+                            static_cast<int>(error_));
+  } else {
+    return CompressionError{};
+  }
+}
+
+CompressionError ZstdCompressContext::SetParameter(int key, uint32_t value) {
+  size_t result = ZSTD_CCtx_setParameter(
+      cctx_.get(), static_cast<ZSTD_cParameter>(key), value);
+  if (ZSTD_isError(result)) {
+    return CompressionError(
+        "Setting parameter failed", "ERR_ZSTD_PARAM_SET_FAILED", -1);
+  }
+  return CompressionError{};
+}
+
+CompressionError ZstdCompressContext::Init() {
+  cctx_.reset(ZSTD_createCCtx());
+  if (!cctx_) {
+    return CompressionError("Could not initialize zstd instance",
+                            "ERR_ZLIB_INITIALIZATION_FAILED",
+                            -1);
+  }
+  return CompressionError{};
+}
+
+CompressionError ZstdCompressContext::ResetStream() {
+  return Init();
+}
+
+void ZstdCompressContext::DoThreadPoolWork() {
+  size_t const remaining =
+      ZSTD_compressStream2(cctx_.get(), &output_, &input_, flush_);
+  if (ZSTD_isError(remaining)) {
+    error_ = ZSTD_getErrorCode(remaining);
+    error_code_string_ = ZstdStrerror(error_);
+    error_string_ = ZSTD_getErrorString(error_);
+    return;
+  }
+}
+
+CompressionError ZstdDecompressContext::SetParameter(int key, uint32_t value) {
```

Comment:
Same here, the value for `ZSTD_DCtx_setParameter()` is an `int`.

Reply:
Thanks, updated to use `int` here.